### DISCUSSION TITLE
How do I configure Fuseki to only serve the SPARQL endpoint for datasets

### DISCUSSION THREAD

**Discussion by stuchalk** [apache/jena#1823] [General]
I am using Fuseki 4.4.0 as service in rhel and I cannot get the configuration so that it disables upload and edit and shows the screen below.  I know this is a noob question but I have spent many hours trying to get this to work.  I realize this is probably not the right place for this question but could not find anywhere better.

![Screen Shot 2023-03-31 at 6 45 49 AM](https://user-images.githubusercontent.com/540150/229100726-7d49844a-55a1-4701-9305-d776dd417ad4.png)

Thanks, [Stuart](https://github.com/stuchalk)

**Comment by afs** [comment]
Specific configuration is by using a configuration file.

See [jena-fuseki2/examples/config-4-endpoint-sparql.ttl](https://github.com/apache/jena/blob/main/jena-fuseki2/examples/config-4-endpoint-sparql.ttl).

It configures query and update on "/someDataset/sparql/".

Adding data requires "gsp-rw" (SPARQL graph store protocol, read and write operations).

The default configuration is illustrated in [jena-fuseki2/examples/config-1-mem.ttl](https://github.com/apache/jena/blob/main/jena-fuseki2/examples/config-1-mem.ttl).

There are other examples in that directory.

Please show what you have tried.

**Comment by stuchalk** [comment]
Andy

Thanks for your quick reply.  In retrospect, I should have detailed all that I have tried. As a developer myself I understand the frustration when someone asks a basic question when there is plenty of documentation available.

One follow up question and a comment.

In terms of the configuration file examples referenced above are these the config.ttl file that goes in fuseki-base, or the dataset specific files that go in 'configurations', or both?  The setup I have is different (inherited from someone else) and it may be my issues are related to the install being setup incorrectly in the first place.

I am a visual learner.  Are there any screencasts of the install process available?  I have looked on YouTube and most are old and not very educational.

Regards, Stuart
https://github.com/stuchalk

**Reply by afs** [reply]
Configuration files are also found in `run/configuration/` - and, up to current version, a fair amount of harmless junk - look for the `function:endpoint` declarations.

**Comment by stuchalk** [comment]
Thanks.
